cmd/statusd-cli/repl.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// REPL implements the read-eval-print loop for the commands
+// to be sent to statusd.
+type REPL struct {
+	host string
+}
+
+// NewREPL creates a REPL instance communicating with the
+// addressed statusd.
+func NewREPL(host string) *REPL {
+	return &REPL{
+		host: host,
+	}
+}
+
+// Run operates the loop to read a command and its arguments,
+// execute it via the client, and print the result.
+func (r *REPL) Run() error {
+	var conn net.Conn
+	var reader *bufio.Reader
+	var writer *bufio.Writer
+	var err error
+	input := bufio.NewReader(os.Stdin)
+	connect := true
+	for {
+		// Connect first time and after connection errors.
+		if connect {
```

Comment:
If it happens once why it can't be done before `for` loop?

Reply:
It always connects once (has been done before loop before). But this way it also reconnects in case of detected connection errors later.

Reply:
Ah sorry, I haven't noticed it gets reset if there was an error.